### PR TITLE
Remove setting clusterIP type for dnsRecord from Schema

### PR DESCRIPTION
**Pull request by orangedeng**
Related issue https://github.com/rancher/rancher/issues/14650
This logic is moved to store for now.

### CODE REVIEW

File: apis/project.cattle.io/v3/schema/schema.go [hunk 1]
```diff
@@ -584,7 +584,6 @@ func addServiceOrDNSRecord(dns bool) types.SchemasInitFunc {
 		schemas = schemas.
 			AddMapperForType(&Version, v1.ServiceSpec{},
 				&m.Move{From: "externalName", To: "hostname"},
```
Comment (alena1108) [inline apis/project.cattle.io/v3/schema/schema.go]: shall we also make clusterIP field non-updatable? there is norman `noupdate` tag

File: apis/project.cattle.io/v3/schema/schema.go [hunk 2]
```diff
@@ -584,7 +584,6 @@ func addServiceOrDNSRecord(dns bool) types.SchemasInitFunc {
...
-				&ServiceSpecMapper{},
```
Comment (alena1108) [inline apis/project.cattle.io/v3/schema/schema.go]: is service spec mapper used anywhere else? If not, lets clean it up

File: apis/project.cattle.io/v3/schema/schema.go (hunk 1 above)
Comment (orangedeng) [reply apis/project.cattle.io/v3/schema/schema.go]: It is non-updatable already in https://github.com/rancher/types/blob/master/apis/project.cattle.io/v3/schema/schema.go#L625

**Comment by orangedeng** [comment]
Updated

**Pull request merged**